conbench/api/runs.py

```diff
@@ -142,72 +143,64 @@ def _search_for_baseline_run(
         )
 
     try:
-        ancestor_commits = (
-            baseline_commit.commit_ancestry_query.order_by(s.desc("commit_order"))
-            .limit(commit_limit)
-            .subquery()
-        )
+        commit_query = baseline_commit.commit_ancestry_query.order_by(
+            s.desc("commit_order")
+        ).limit(commit_limit)
     except CantFindAncestorCommitsError as e:
         return _CandidateBaselineSearchResult(
             error=f"could not find the baseline commit's ancestry because {e}"
         )
 
+    commits = commit_query.all()
+    commit_ids = [commit.ancestor_id for commit in commits]
+    commit_hashes = [commit.ancestor_hash for commit in commits]
```

Comment:
In terms of code readability and head scratchability it might be worth explaining why we use `[commit.ancestor_id for commit in commits]` instead of `[commit.id for commit in commits]` :) Same for hash.

There probably is a good reason, it looks like one of those really important off-by-one considerations!

Reply:
Maybe this is mainly about the `commits`  variable name vs `commit_ids` which kind of suggests that they both refer to the same set/collection of commits. But that's probably not true. Maybe we want `commit_ids` and then `ancestor_commit_ids`? Well. :shrug: 

And not important but warming up the single-letter var conversation: in list comprehensions I am super cool with single letter loop vars, e.g. `[c.id for c in commits]`.

Sorry, moving on :D.

Reply:
yeah, good questions! This confusion comes from the fact that `commits` is the result of executing the [commit_ancestry_query](https://github.com/conbench/conbench/blob/9ca3c6d84ec3a0ab47c2f3368b4bbe2dcca5479b/conbench/entities/commit.py#L180), which prefixes its colnames with `ancestor_` to disambiguate them from when we join it to the actual `Commit` table.

But we don't join it anymore in this query, and I'm planning to remove the join in the z-score query in my next PR. So I'll clean this up then.